pkg/eval/mods/file/file.go

```diff
@@ -0,0 +1,100 @@
+// File module which is an equivalent of POSIX like commands.
+// Checkout issule number 1263 for more information.
+
+package file
+
+import (
+	"os"
+	"src.elv.sh/pkg/eval"
+	"src.elv.sh/pkg/eval/vals"
+)
+
+// elvdoc:fn fopen
+// ```elvish
+// fopen $filename
+// ```
+// Opens the file for reading.
+// File should be closed after opening using fclose
+// Example
+//  ```elvish-transcript
+// ~> cat a.txt
+// Some text in file
+// f = (fopen a.txt)
+// ~> cat < $f
+// Some text in file
+// ~> fclose $f
+// ```
+func fopen(name string) (vals.File, error) {
+	return os.Open(name)
+}
+
+// elvdoc: fn fclose
+// flcose is used to close an already opened file.
+// files are opened for reading using fopen
+// Will not accept string name of the file
+// ```elvish
+// fclose $fileptr
+// ```
+func fclose(f vals.File) error {
+	return f.Close()
+}
+
+//elvdoc:fn pipe
```

Comment:
I think you're attempting to do too much with this section of the change and would recommend omitting the `pipe`, `prclose` and `pwclose` commands from this module. They can be introduced by a subsequent change.